metagraph/src/annotation/taxonomy/tax_classifier.cpp

```diff
@@ -0,0 +1,299 @@
+#include "tax_classifier.hpp"
+
+#include <cmath>
+#include <string>
+#include <vector>
+
+#include "annotation/representation/annotation_matrix/annotation_matrix.hpp"
+#include "common/unix_tools.hpp"
+#include "common/utils/string_utils.hpp"
+
+#include "common/logger.hpp"
+
+namespace mtg {
+namespace annot {
+
+using mtg::common::logger;
+
+bool TaxonomyBase::assign_label_type(const std::string &sample_label) {
+    if (utils::starts_with(sample_label, "gi|")) {
+        // e.g.   >gi|1070643132|ref|NC_031224.1| Arthrobacter phage Mudcat, complete genome
+        label_type = GEN_BANK;
+        return true;
+    } else if (utils::starts_with(utils::split_string(sample_label, ":")[1], "taxid|")) {
+        // e.g.   >kraken:taxid|2016032|NC_047834.1 Alteromonas virus vB_AspP-H4/4, complete genome
+        label_type = TAXID;
+        return false;
+    }
+
+    logger->error("Can't determine the type of the given label {}. "
+                  "Make sure the labels are in a recognized format.", sample_label);
+    exit(1);
+}
+
+bool TaxonomyBase::get_taxid_from_label(const std::string &label, TaxId *taxid) const {
+    if (label_type == TAXID) {
+        *taxid = std::stoul(utils::split_string(label, "|")[1]);
+        return true;
+    } else if (TaxonomyBase::label_type == GEN_BANK) {
+        std::string acc_version = get_accession_version_from_label(label);
+        if (not accversion_to_taxid_map.count(acc_version)) {
+            return false;
+        }
+        *taxid = accversion_to_taxid_map.at(acc_version);
+        return true;
+    }
+
+    logger->error("Error: Could not get the taxid for label {}", label);
+    exit(1);
+}
```

Comment:
I'm confused by these helpers.

You are setting `label_type` somewhere else, then it's used here. So it requires a certain sequence of these calls right?
What is the point to separate these functions then?

Reply:
I will need to use this `label_type` from different enterpoints from the CLI cmd. Currently, it is computed in the `TaxonomyClsAnno` constructor and will be further used inside `tax_class(sequence)` method where we will need to iterate the column labels and take the associated taxids. If you think that it would be more clear, I can move it as part of the `tax_class` and run it multiple times (should be fast, I guess).